src/samples/OpenChannelMessageThreading.js

```diff
@@ -0,0 +1,718 @@
+          isOpenThread={state.isOpenThread}
+          onFileInputChange={onFileInputChange} />
+      </Channel>
+      <Threads
+        isOpenThread={state.isOpenThread}
+        openThreads={openThreads}
+        exitThreads={exitThreads}
+        handleDeleteMessage={handleDeleteMessage}
+        updateMessage={updateMessage}
+        threadsParentsMessage={state.threadsParentsMessage}
+      >
+        <MessagesList
+          displayNone={"display-none"}
+          messages={state.threadsMessages}
+          handleDeleteMessage={handleDeleteMessage}
+          updateMessage={updateMessage}
+        />
+        <MessageInput
+          threadsInputClass={"threads-input"}
+          value={state.threadsMessageInputValue}
+          isOpenThread={state.isOpenThread}
+          isThread={true}
+          onChange={onThreadsMessageInputChange}
+          sendMessage={sendThreadMessage}
+          fileSelected={state.threadFile}
+          onFileThreadsInputChange={onFileThreadsInputChange} />
+      </Threads>
+    </>
+  );
+};
+
+// Chat UI Components
+const ChannelList = ({ channels, handleJoinChannel, toggleShowCreateChannel, handleDeleteChannel, toggleChannelDetails }) => {
+  return (
+    <div className='channel-list'>
+      <div className="channel-type">
+        <h1>Open Channels</h1>
+        <button className="channel-create-button" onClick={toggleShowCreateChannel}>Create Channel</button>
+      </div>
+      {
+        channels.map(channel => {
+          const userIsOperator = channel.operators.some((operator) => operator.userId === sb.currentUser.userId)
+          return (
+            <div key={channel.url} className="channel-list-item" >
+              <div className="channel-list-item-name"
+                   onClick={() => { handleJoinChannel(channel.url) }}>
+                {channel.name}
+              </div>
+              {userIsOperator &&
+              <div>
+                <button className="control-button" onClick={() => toggleChannelDetails(channel)}>
+                  <img className="channel-icon" src='/icon_edit.png' />
+
+                </button>
+                <button className="control-button" onClick={() => handleDeleteChannel(channel.url)}>
+                  <img className="channel-icon" src='/icon_delete.png' />
+
+                </button>
+              </div>}
+            </div>);
+        })
+      }
+    </div >);
+}
+
+
+const Channel = ({ currentlyJoinedChannel, handleLeaveChannel, children }) => {
+  if (currentlyJoinedChannel) {
+    return <div className="channel">
+      <ChannelHeader>{currentlyJoinedChannel.name}</ChannelHeader>
+      <div>
+        <button className="leave-channel" onClick={handleLeaveChannel}>Exit Channel</button>
+      </div>
+      <div>{children}</div>
+    </div>;
+
+  }
+  return <div className="channel"></div>;
+
+}
+
+const Threads = ({ isOpenThread, exitThreads, children, threadsParentsMessage, handleDeleteMessage, updateMessage}) => {
```

Comment:
Very small but do we not want to use "Thread" singular here and in this component